### PR TITLE
[WASH-1055] Tests for identity login audits

### PR DESCRIPTION
**Pull request by carrala**
- tests the cases where a username, userID or a clientID is provided

### CODE REVIEW

File: identityClient/identityClient.go [hunk 1]
```diff
@@ -664,6 +666,20 @@ func (c *E3dbIdentityClient) IdentityLogin(ctx context.Context, realmName string
 	return identity, err
 }
 
+// InternalCreateIdentityLoginAudit creates an audit of the login attempt returning all the information about the login
+// attempt and error (if any)
+func (c *E3dbIdentityClient) InternalCreateIdentityLoginAudit(ctx context.Context, params InternalIdentityLoginAudit) (*InternalIdentityLoginAudit, error) {
+	var loginAudit *InternalIdentityLoginAudit
+	path := c.Host + internalIdentityServiceBasePath + fmt.Sprintf("/%s/%s", auditResourceName, loginResourceName)
+	// might want to update this so that it takes a part of the params part ...
```
Comment (rosasck) [inline identityClient/identityClient.go:674]: If you dont want an item in the struct to be sent over to the endpoint dont include the `json:___" then it wont be marshalled!

**Review by rosasck — APPROVED** [review]
SO far so great!!  Dont forget comments and if you want to keep your t.logs you can change them to debug!

**Review by galxy25 — APPROVED** [review]

File: identityClient/api.go [hunk 2]
```diff
@@ -246,6 +246,17 @@ type InternalIdentityStatusStorageClientIdRequest struct {
 	StorageClientID uuid.UUID // The ID of the Identity's Tozny Client ID
 }
 
+type InternalIdentityLoginAudit struct {
```
Comment (rosasck) [inline identityClient/api.go:249]: Add a comment

File: identityClient/identityClient_internal_test.go [hunk 3]
```diff
@@ -276,6 +276,93 @@ func TestInternalIdentityLoginWithAuthenticatedRealmIdentity(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("RegisterIdentity Error: %+v\n", err)
+	}
+	auditParamsUsername := InternalIdentityLoginAudit{
+		RealmDomain: realm.Domain,
+		Username:    identity.Identity.Name,
+		Status:      "fail",
+		RequestType: "test with username",
+	}
+	_, err = identityServiceClient.InternalCreateIdentityLoginAudit(testContext, auditParamsUsername)
```
Comment (rosasck) [inline identityClient/identityClient_internal_test.go:340]: What is the expected response? Would also check that and fail if you dont get it back

**Review by rosasck — APPROVED** [review]
pending checking the return for the test calls

**Pull request merged**